### PR TITLE
Add `story` icon

### PR DESCRIPTION
**Pull request by jordan808**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description
Adds `story` icon as requested by https://github.com/lucide-icons/lucide/issues/119. Closes feather icons issue https://github.com/feathericons/feather/issues/1019.

### Icon use case 

The purpose of this icon is to represent stories, as popularized by Snapchat and Instagram. Use cases include referencing, or enabling users to create their own stories in an app, or for a digital marketing platform that automates the posting of stories as a feature.

### Alternative icon designs 

story-alt

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [ ] I've based them on the following Lucide icons: 
- [x] I've based them on the following design: The Instagram stories icon as seen in the app under story settings -> 
IMG_0300

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/story.svg [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path
```
Comment (ericfennis) [inline icons/story.svg:12]: The paths should not be a compound path (combined path). Design programs like Figma don't like them. Please split them up in separate elements/

**Review by ericfennis — CHANGES_REQUESTED** [review]
Nice work @jordan808!

File: icons/story.svg (hunk 1 above)
Comment (jguddas) [reply icons/story.svg:12]: You can just paste it in https://lucide-studio.vercel.app and press <kbd>Tidy</kbd>.

Comment (jordan808) [reply icons/story.svg:12]: Ahh will do! I've been using SVGOMG as per documentation which combines all the paths, even though I agree that keeping them separate makes it easier to make changes later. That link would be great in the docs @jguddas

Comment (jordan808) [reply icons/story.svg]: I split up the paths and also made some minor adjustments to have an almost exact 2px spacing between dashes (hard with ellipses).

**Review by ericfennis — APPROVED** [review]
Nice work, LGTM

File: icons/story.svg [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M12 2a10 10 0 0 1 7.38 16.75l-.06.06" />
+  <path d="M12 8v8" />
+  <path d="M16 12H8" />
+  <path d="M2.42 9.11a10 10 0 0 0-.42 3" />
+  <path d="m2.8 15.92.03.07a9.95 9.95 0 0 0 2.43 3.4" />
+  <path d="M4.1 5.88a9.449 9.449 0 0 1 .58-.68" />
+  <path d="M8.64 21.42A10.27 10.27 0 0 0 12 22a9.907 9.907 0 0 0 4.28-.96" />
```
Comment (jguddas) [inline icons/story.svg:18]:
```suggestion
  <path d="M11.998 2a10 10 0 0 1 7.38 16.75" />
  <path d="M12 8v8" />
  <path d="M16 12H8" />
  <path d="M2.501 8.875a10 10 0 0 0-.5 3" />
  <path d="M2.829 15.986a10 10 0 0 0 2.43 3.4" />
  <path d="M4.636 5.235a10 10 0 0 1 .891-.857" />
  <path d="M8.644 21.42a10 10 0 0 0 7.631-.38" />
```

File: icons/story.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "jordan808"
```
Comment (jguddas) [inline icons/story.json:4]:
```suggestion
    "jordan808",
    "jguddas"
```

**Review by jguddas — CHANGES_REQUESTED** [review]

File: icons/story.svg (hunk 2 above)
Comment (jordan808) [reply icons/story.svg]: Looks good and added. Just curious, but how did you refine the SVG? Looking to get better at making things pixel-perfect 🙏

**Comment by jguddas** [comment]
What about the name?

It should be something like `circle-[something that describes a dissapering dash]-plus`.

https://lucide.dev/guide/design/icon-design-guide#naming-conventions

**Comment by jordan808** [comment]
> What about the name?
> 
> It should be something like `circle-plus-tapered-dash`.
> 
> https://lucide.dev/guide/design/icon-design-guide#naming-conventions


I assumed story was already agnostic but you are right that it would have to be changed. In this case, it should also be included in the "shapes" category. Your suggestion would also follow the conventions of the other dashed circles like `circle-dot-dashed`: circle-dot-dashed

Should I update the name and category?

**Comment by karsa-mistmere** [comment]
> I assumed story was already agnostic but you are right that it would have to be changed.

I think @jguddas is talking about naming convention no. 3:
> Icons should be named for what they depict rather than their use case or what they represent.

> `circle-plus-tapered-dash`

I don't know, I feel like this might be a bit misleading, as if the plus itself was tapered and/or dashed.

How about `circle-[adjective]-plus`, [adjective] being one of these?

- 😄 `fading`
- 🎉 `dissolving`
- 😕 `tapering`
- ❤️ `vanishing`
- 🚀 `diminishing`
- 👀 `dissipating`

**Comment by jordan808** [comment]
Ahh good point @karsa-mistmere ! Do you have a preference? I like the idea of `circle-fading-plus`

**Review by jguddas — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]

**Pull request merged**